erpnext/buying/doctype/purchase_order/purchase_order.py

```diff
@@ -295,7 +295,10 @@ def update_receiving_percentage(self):
 		for item in self.items:
 			received_qty += item.received_qty
 			total_qty += item.qty
-		self.db_set("per_received", flt(received_qty/total_qty) * 100, update_modified=False)
+		try:
+			self.db_set("per_received", flt(received_qty/total_qty) * 100, update_modified=False)
+		except:
```

Comment:
This will catch all exceptions - even ones you didn't intend to. To avoid unexpected behavior, you should catch the `ZeroDivisionError` explicitly.

Alternatively, if you're only expecting problems for zero-quantity purchase orders, it might be more readable to replace the `try..except` with an `if..else` check on the `total_qty`.